server/athenian/api/experiments/preloading/cache.py

```diff
@@ -0,0 +1,259 @@
+import asyncio
+from typing import Dict, List, Optional, Union
+
+import pandas as pd
+import sqlalchemy as sa
+from sqlalchemy import select
+
+from athenian.api.async_utils import read_sql_query
+from athenian.api.typing_utils import DatabaseLike
+
+
+def _humanize_size(v):
+    unit = ""
+    for u in ["B", "KB", "MB", "GB"]:
+        unit = u
+        if v < 1024:
+            break
+        v = v / 1024
+
+    return f"{round(v, 2)} {unit}"
+
+
+class CachedDataFrame:
+    """A DataFrame that is loaded and can be refreshed from the database.
+
+    This can be read as a normal `pandas.DataFrame`.
+
+    """
+
+    def __init__(
+        self,
+        id_: str,
+        db: DatabaseLike,
+        cols: Optional[List[sa.Column]] = None,
+        timestamp_cols: Optional[List[sa.Column]] = None,
+        categorical_cols: Optional[List[sa.Column]] = None,
+        identifier_cols: Optional[List[sa.Column]] = None,
+        filtering_clause: Optional[sa.sql.ClauseElement] = None,
+        debug_memory: Optional[bool] = False,
```

Comment:
Instead of making it a ctor arg, we'd rather reference the log level of `athenian.api.CachedDataFrame`, given that it's as easy to change it as to define `flog_athenian.api.CachedDataFrame=DEBUG` environment variable.

Reply:
It makes a lot of sense, I wasn't aware that it was that easy to change the log level of each logger, that's cool 👍 

BTW TIL what `ctor` means 😂

Reply:
Sure, we can keep the arg if you like it :smile: 

```python
from athenian.api import metadata

class Foo:
    _log = logging.getLogger(f"{metadata.__package__}.Foo")

    def __init__(self):
        debug_mode = self._log.isEnabledFor(logging.DEBUG)
```